.github/workflows/codeql.yml

```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: '15 3 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go', 'python' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+        # this covers a larger portion of checks but are experimental right now
+        queries: security-and-quality
+    - run: |
+       make build
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
+
```

Comment:
Not sure if this will work - does it work on compiled code? The make/run instructions are different for our code as far as I remember.

Reply:
it does work on program files and compiled binaries. More about this here: https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-the-codeql-workflow-for-compiled-languages#about-autobuild-for-codeql

Reply:
and I did check it with Tasos and also cross  checked the Makefile, `make build` is how we code, right?

Reply:
Still not sure that we should build manually

Reply:
@ankurdotb @anikitinDSR can you guys share what you think?